pkg/model/instance.go

```diff
@@ -34,38 +33,20 @@ type Linked struct {
 	DestinationInstance   Instance `json:"destinationInstance"`
 }
 
-func (i Instance) FindRequiredParameter(name string) (InstanceRequiredParameter, error) {
-	for _, parameter := range i.RequiredParameters {
-		if parameter.StackRequiredParameterID == name {
+func (i Instance) FindParameter(name string) (InstanceParameter, error) {
+	for _, parameter := range i.Parameters {
+		if parameter.StackParameterID == name {
 			return parameter, nil
 		}
 	}
-	return InstanceRequiredParameter{}, fmt.Errorf("required parameter not found: %s", name)
+	return InstanceParameter{}, fmt.Errorf("required parameter not found: %s", name)
 }
 
-func (i Instance) FindOptionalParameter(name string) (InstanceOptionalParameter, error) {
-	for _, parameter := range i.OptionalParameters {
-		if parameter.StackOptionalParameterID == name {
-			return parameter, nil
-		}
-	}
-	return InstanceOptionalParameter{}, fmt.Errorf("optional parameter not found: %s", name)
-}
-
-type InstanceRequiredParameter struct {
-	InstanceID uint `json:"-" gorm:"primaryKey"`
-	// TODO: Rename StackRequiredParameterID to Name
-	StackRequiredParameterID string                 `json:"name" gorm:"primaryKey"`
-	StackRequiredParameter   StackRequiredParameter `json:"-" gorm:"foreignKey:StackRequiredParameterID,StackName; references:Name,StackName; constraint:OnUpdate:CASCADE,OnDelete:CASCADE"`
-	StackName                string                 `json:"-"`
-	Value                    string                 `json:"value"`
-}
-
-type InstanceOptionalParameter struct {
+type InstanceParameter struct {
 	InstanceID uint `json:"-" gorm:"primaryKey"`
-	// TODO: Rename StackOptionalParameterID to Name
-	StackOptionalParameterID string                 `json:"name" gorm:"primaryKey"`
-	StackOptionalParameter   StackOptionalParameter `json:"-" gorm:"foreignKey:StackOptionalParameterID,StackName; references:Name,StackName; constraint:OnUpdate:CASCADE,OnDelete:CASCADE"`
-	StackName                string                 `json:"-"`
-	Value                    string                 `json:"value"`
+	// TODO: Rename StackParameterID to Name
```

Comment:
does it make sense to do this as well with this PR?

Reply:
No, after merging the database service I didn't fix all the relations which obviously should be done but it's beyond the scope of this PR